src/components/avatar/index.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { FC } from 'react';
+
+import { BoxProps, useRecipe } from '@chakra-ui/react';
+
+import { avatarRecipe } from 'src/themes/shared/recipes/avatar';
+
+import { AvatarIcon, AvatarWithNotificationIcon } from '../icons';
+
+export type AvatarProps = {
+  withNotification?: boolean;
+  color?: BoxProps['color'];
+};
+
+export const Avatar: FC<AvatarProps> = ({ withNotification, color }) => {
+  const recipe = useRecipe({ recipe: avatarRecipe});
```

Comment:
Since we're exposing the recipe via theme I'd go for a `key` based consumption:
```suggestion
  const recipe = useRecipe({ key: 'avatar' });
```